torch/nn/modules/transformer.py

```diff
@@ -314,20 +334,8 @@ def forward(
                 output = torch._nested_tensor_from_mask(output, src_key_padding_mask.logical_not(), mask_check=False)
                 src_key_padding_mask_for_layers = None
 
-        # Prevent type refinement
-        make_causal = (is_causal is True)
-
-        if is_causal is None:
-            if mask is not None:
-                sz = mask.size(0)
-                causal_comparison = torch.triu(
-                    torch.ones(sz, sz, device=mask.device) * float('-inf'), diagonal=1
-                ).to(mask.dtype)
-
-                if torch.equal(mask, causal_comparison):
-                    make_causal = True
-
-        is_causal = make_causal
+        seq_len = None if (src.is_nested or not hasattr(self, "seq_len_pos")) else src.size(self.seq_len_pos)
```

Comment:
This is not really a "PR review" comment but to aid my growing understanding of Transformer related nn APIs, how does the causal mask interplay with a nested `src`?

Reply:
`src.size()` fails on nested tensors.  you can't know what the user had before because the sequence-length is no longer a property of the batch, but each batch element hasd a length.  However, the mask would still correspond to whatever the original size was -- which you can't reconstruct from the nested tensor alone